openimis-config-vite.js

```diff
@@ -0,0 +1,126 @@
+const fs = require("fs");
+const pkg = require("./package.json");
+
+function processLocales(config) {
+  var locales = fs.createWriteStream("./src/locales.jsx");
+  let localeByLang = config["locales"].reduce((lcs, lc) => {
+    lc.languages.forEach((lg) => (lcs[lg] = lc.intl));
+    return lcs;
+  }, {});
+  let filesByLang = config["locales"].reduce((fls, lc) => {
+    lc.languages.forEach((lg) => (fls[lg] = lc.fileNames));
+    return fls;
+  }, {});
+  locales.write(`export const locales = ${JSON.stringify(config["locales"].map((lc) => lc.intl))}`);
+  locales.write(`\nexport const fileNamesByLang = ${JSON.stringify(filesByLang)}`);
+  locales.write(`/* eslint import/no-anonymous-default-export: [2, {"allowObject": true}] */`);
+  locales.write(`\nexport default ${JSON.stringify(localeByLang)}`);
+}
+
+function getConfig() {
+  // Try to get the configuration from the args
+  if (process.argv[2]) {
+    console.log(`  load configuration from '${process.argv[2]}'`);
+    return JSON.parse(fs.readFileSync(process.argv[2], "utf-8"));
+  } else if (process.env.OPENIMIS_CONF_JSON) {
+    console.log(`  load configuration from env`);
+    return JSON.parse(process.env.OPENIMIS_CONF_JSON);
+  } else if (fs.existsSync("./openimis.json")) {
+    console.log(`  load configuration from ./openimis.json`);
+    return JSON.parse(fs.readFileSync("./openimis.json", "utf-8"));
+  } else {
+    throw new Error(
+      "No configuration file found. Please provide a configuration in the CLI or in the OPENIMIS_CONF_JSON environment variable",
+    );
+  }
+}
+
+function processModules(modules) {
+  const stream = fs.createWriteStream("./src/modules.jsx");
+
+  stream.write(`
+export const packages = [
+  ${modules.map(({ moduleName }) => `"${moduleName}"`).join(",\n  ")}
+];
+
+`);
+
+  stream.write(`
+export function loadModules(cfg = {}) {
+  const loadedModules = [];
+${modules
+.map(({ name, logicalName, moduleName }) => {
+return `
+  try {
+    // Using dynamic import for Vite compatibility
+    import("${moduleName}")
+      .then(module => {
+        loadedModules.push(module.${name ?? "default"}(cfg["${logicalName}"] || {}));
+      })
+      .catch(error => {
+        console.error(\`Failed to load module "${moduleName}". Error: \${error}\`);
+        alert(\`Failed to load module "${moduleName}". More details can be found in the developer console.\`);
+      });
+  } catch (error) {
+    console.error(\`Failed to import module "${moduleName}". Error: \${error}\`);
+    alert(\`Failed to import module "${moduleName}". More details can be found in the developer console.\`);
+  }
+`;
+})
+.join("")}
+  return loadedModules;
+}
+`);
+
+  stream.end();
+}
+
+function main() {
```

Comment:
If I understand correctly, this is effectively doing what's currently done in https://github.com/openimis/openimis-fe_js/blob/develop/openimis-config.js In that case, as part of the migration should openimis-config.js be removed? Are there other similar cleanups that need to happen? I see lots of addition with this PR but no removal – please be sure to clean up all CRA related scripts for a clean migration.